packages/hix/lib/Hix/Path.hs

```diff
@@ -23,3 +28,18 @@ inferRoot = do
 rootDir :: Maybe (Path Abs Dir) -> M (Path Abs Dir)
 rootDir =
   maybe inferRoot pure
+
+resolvePathSpecWith :: 
+  (Path Abs Dir -> FilePath -> IO (Path Abs t)) ->
+  PathSpec t ->
+  M (Path Abs t)
+resolvePathSpecWith resolver pathSpec =
+  appContext [exon|resolving path spec #{showHP pathSpec}|] $ do
+    AppResources {cwd} <- M ask
+    liftE $ resolvePathSpec resolver cwd pathSpec
+
+resolvePathSpecDir :: PathSpec Dir -> M (Path Abs Dir)
+resolvePathSpecDir = resolvePathSpecWith resolveDir
+
+resolvePathSpecFile :: PathSpec File -> M (Path Abs File)
+resolvePathSpecFile = resolvePathSpecWith resolveFile
```

Comment:
oh I forgot: would be nice to turn this into a class, so we'd have `resolvePathSpec :: PathSpec t -> M (Path Abs t)`